server/services/generateReportService.ts

```diff
@@ -14,99 +14,208 @@
  */
 
 //@ts-ignore
-import { Legacy, Logger, PluginInitializerContext } from 'kibana';
+import { Legacy, PluginInitializerContext } from 'kibana';
+import { Logger } from '../../../../src/core/server/logging';
 import { Server, Request, ResponseToolkit } from 'hapi';
 import { RequestParams } from '@elastic/elasticsearch';
 import puppeteer from 'puppeteer';
-import imagesToPdf from 'images-to-pdf';
-import fs from 'fs';
 import { v1 as uuidv1 } from 'uuid';
 import { ServerResponse } from '../models/types';
 import { CLUSTER, FORMAT } from '../utils/constants';
+import { SearchResponse, GetResponse } from 'elasticsearch';
+import { Readable } from 'stream';
 
 type ElasticsearchPlugin = Legacy.Plugins.elasticsearch.Plugin;
 
 export default class GenerateReportService {
   esDriver: ElasticsearchPlugin;
-  private readonly log: Logger;
+  logger: Logger;
 
   constructor(
     private readonly initializerContext: PluginInitializerContext,
     esDriver: ElasticsearchPlugin
   ) {
     this.esDriver = esDriver;
-    this.log = this.initializerContext.logger.get();
+    this.logger = this.initializerContext.logger.get();
   }
 
   report = async (req: Request, h: ResponseToolkit): Promise<any> => {
     try {
       const {
         url,
         itemName,
+        source,
         reportFormat,
         windowWidth = 1200,
         windowLength = 800,
       } = req.payload as {
         url: string;
         itemName: string;
+        source: string;
         reportFormat: string;
```

Comment:
Create whitelisting for `reportFormat` to ensure no other formats are supported or being accepted from server end. Client can manipulate anything.

Reply:
I assume you mean input validation. I have done that with `Joi` here https://github.com/opendistro-for-elasticsearch/kibana-reports/blob/a9c122fcf20f57ce48c4675cca6089e1eae4ac69/server/routes/generateReport.ts#L37